convert2rhel/unit_tests/checks_test.py

```diff
@@ -730,6 +730,20 @@ def test_bad_kernel_fingerprint(
     )
     monkeypatch.setattr(checks, "get_pkg_fingerprint", get_pkg_fingerprint_mocked)
     assert _bad_kernel_package_signature(kernel_release) == exp_return
+    run_subprocess_mocked.assert_called_with(
+        ["rpm", "-qf", "--qf", "%{NAME}", "/boot/vmlinuz-%s" % kernel_release], print_output=False
+    )
+
+
+def test_kernel_not_installed(caplog, monkeypatch):
+    run_subprocess_mocked = mock.Mock(spec=run_subprocess, return_value=(" ", 1))
+    monkeypatch.setattr(checks, "run_subprocess", run_subprocess_mocked)
+    assert _bad_kernel_package_signature("4.18.0-240.22.1.el8_3.x86_64")
+    log_message = (
+        "The booted kernel /boot/vmlinuz-4.18.0-240.22.1.el8_3.x86_64 is not owned by any installed package."
+        " It needs to be owned by a package signed by CentOS."
+    )
+    assert log_message in caplog.text
```

Comment:
Just a side note, I tried to run this test locally and it failed to me. 

```bash
(.venv3) [r0x0d@fedora convert2rhel]$ pytest -sk test_kernel_not_installed
=============================================================== test session starts ================================================================
platform linux -- Python 3.10.4, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/r0x0d/Workspace/r0x0d/convert2rhel, configfile: pytest.ini, testpaths: convert2rhel/unit_tests
plugins: cov-3.0.0
collected 344 items / 343 deselected / 1 selected                                                                                                  

convert2rhel/unit_tests/checks_test.py F

===================================================================== FAILURES =====================================================================
____________________________________________________________ test_kernel_not_installed _____________________________________________________________

caplog = <_pytest.logging.LogCaptureFixture object at 0x7f2ab0015510>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f2ab0015600>

    def test_kernel_not_installed(caplog, monkeypatch):
        run_subprocess_mocked = mock.Mock(spec=run_subprocess, return_value=(" ", 1))
        monkeypatch.setattr(checks, "run_subprocess", run_subprocess_mocked)
>       assert _bad_kernel_package_signature("4.18.0-240.22.1.el8_3.x86_64")

convert2rhel/unit_tests/checks_test.py:741: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

kernel_release = '4.18.0-240.22.1.el8_3.x86_64'

    def _bad_kernel_package_signature(kernel_release):
        """Return True if the booted kernel is not signed by the original OS vendor, i.e. it's a custom kernel."""
        vmlinuz_path = "/boot/vmlinuz-%s" % kernel_release
        kernel_pkg, return_code = run_subprocess(["rpm", "-qf", "--qf", "%{NAME}", vmlinuz_path], print_output=False)
        logger.debug("Booted kernel package name: {0}".format(kernel_pkg))
    
        if return_code == 1:
>           os_vendor = system_info.name.split()[0]
E           AttributeError: 'NoneType' object has no attribute 'split'

convert2rhel/checks.py:430: AttributeError
---------------------------------------------------------------- Captured log call -----------------------------------------------------------------
FILE     convert2rhel.checks:checks.py:427 Booted kernel package name:
================================================================= warnings summary =================================================================
../../../../../usr/lib64/python3.10/subprocess.py:956
../../../../../usr/lib64/python3.10/subprocess.py:956
../../../../../usr/lib64/python3.10/subprocess.py:956
  /usr/lib64/python3.10/subprocess.py:956: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
    self.stdout = io.open(c2pread, 'rb', bufsize)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================= short test summary info ==============================================================
FAILED convert2rhel/unit_tests/checks_test.py::test_kernel_not_installed - AttributeError: 'NoneType' object has no attribute 'split'
================================================== 1 failed, 343 deselected, 3 warnings in 0.33s ===================================================
```

Maybe `system_info` is being initialized in other tests in the `checks_test.py` and thus giving the impression that this one is passing normally in the CI?

Reply:
What do you think @abadger?

Reply:
I didn't trace the unittest code but I think @r0x0d is correct.  Monkeypatching system_info.name inside of this test function is probably what we should do.

Reply:
Another easy approach to fix this is to call the @centos8 decorator in the test function, for example. This way we have the system_info mocked and ready to go.

Reply:
I have made the change to this comment as well!

Reply:
I have taken care of this comment! thank you both for the input!